presto-thrift-connector/src/test/java/com/facebook/presto/connector/thrift/clientproviders/MockConnectedThriftServiceProvider.java

```diff
@@ -0,0 +1,155 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.connector.thrift.clientproviders;
+
+import com.facebook.presto.connector.thrift.api.PrestoThriftId;
+import com.facebook.presto.connector.thrift.api.PrestoThriftNullableColumnSet;
+import com.facebook.presto.connector.thrift.api.PrestoThriftNullableSchemaName;
+import com.facebook.presto.connector.thrift.api.PrestoThriftNullableTableMetadata;
+import com.facebook.presto.connector.thrift.api.PrestoThriftNullableToken;
+import com.facebook.presto.connector.thrift.api.PrestoThriftPageResult;
+import com.facebook.presto.connector.thrift.api.PrestoThriftSchemaTableName;
+import com.facebook.presto.connector.thrift.api.PrestoThriftServiceException;
+import com.facebook.presto.connector.thrift.api.PrestoThriftSplitBatch;
+import com.facebook.presto.connector.thrift.api.PrestoThriftTupleDomain;
+import com.facebook.presto.connector.thrift.location.HostLocationHandle;
+import com.facebook.presto.spi.HostAddress;
+import com.google.common.collect.ImmutableList;
+import com.google.common.util.concurrent.ListenableFuture;
+
+import java.util.List;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import static com.facebook.presto.spi.HostAddress.fromParts;
+
+public class MockConnectedThriftServiceProvider
```

Comment:
Don't call it mock. It's a test implementation. The convention in Presto is to prefix them with "Testing..". See examples in the code.